pipeline/seq_retrieval/src/main.py

```diff
@@ -0,0 +1,110 @@
+"""
+Main module serving the CLI for PAVI sequence retrieval.
+
+Retrieves multiple sequence regions and returns them as one chained sequence.
+"""
+import click
+import typing
+import json
+
+import data_mover.data_file_mover as data_file_mover
+from seq_region import SeqRegion, chain_seq_region_seqs
+
+def validate_strand_param(ctx, param, value) -> typing.Literal['+', '-']:
+    """
+    Processes and normalises the value of click input argument `strand`.
+
+    Returns:
+        A normalised version of strings representing a strand: '-' or '+'
+
+    Raises:
+        click.BadParameter: If an unrecognised string was provided.
+    """
+    POS_CHOICES = ['+', '+1', 'pos']
+    NEG_CHOICES = ['-', '-1', 'neg']
+    if   value in POS_CHOICES:
+        return '+'
+    elif value in NEG_CHOICES:
+        return '-'
+    else:
+        raise click.BadParameter(f"Must be one of {POS_CHOICES} for positive strand, or {NEG_CHOICES} for negative strand.")
+
+def process_seq_regions_param(ctx, param, value) -> typing.List[typing.Dict[str,typing.Any]]:
```

Comment:
Small points (being super picky :-)) but it would be nice to have the typing on the inputs here.
Also would be easier to read if you import the specific types you need and then use those:-
so instead of 

import typing
  something: typing.Dict = {}

instead
import from typing Dict
  something: Dict = {}

It is just easier to read especially for longer types or Lists of things, but as I said very minor point.

Reply:
Minor but valid points, now fixed 👍